cpg_workflows/stages/aip.py

```diff
@@ -0,0 +1,294 @@
+        )
+
+        return self.make_outputs(dataset, data=expected_d, jobs=py_job)
+
+
+@stage(required_stages=[GeneratePanelData])
+class QueryPanelapp(DatasetStage):
+    """
+    query PanelApp for up-to-date gene lists
+    """
+
+    def expected_outputs(self, dataset: Dataset) -> dict[str, Path]:
+        return {'panel_data': dataset.prefix() / DATED_FOLDER / 'panelapp_data.json'}
+
+    def queue_jobs(self, dataset: Dataset, inputs: StageInput) -> StageOutput:
+
+        job = get_batch().new_job('query panel data')
+        STANDARD.set_resources(job, ncpu=1, mem_gb=1.0)
+        job.image(image_path('aip'))
+
+        # auth and copy env
+        authenticate_cloud_credentials_in_job(job)
+        copy_common_env(job)
+
+        hpo_panel_json = inputs.as_path(
+            target=dataset, stage=GeneratePanelData, key='hpo_panels'
+        )
+        expected_out = self.expected_outputs(dataset)
+        job.command(
+            f'python3 reanalysis/query_panelapp.py '
+            f'--panels {str(hpo_panel_json)} '
+            f'--out_path {str(expected_out["panel_data"])}'
+        )
+
+        return self.make_outputs(dataset, data=expected_out, jobs=job)
+
+
+def query_for_latest_mt(dataset: str) -> str:
+    """
+    query for the latest MT for a dataset
+    Args:
+        dataset (str):
+    Returns:
+        str, the path to the latest MT
+    """
+    my_query = gql(
+        """
+        query MyQuery($dataset: String!) {
+            project(name: $dataset) {
+                analyses(type: {eq: "custom"}, status: {eq: COMPLETED}) {
+                    output
+                }
+            }
+        }
+    """
+    )
+    result = query(my_query, variables={'dataset': dataset})
+    seq_type_exome = get_config()['workflow'].get('sequencing_type') == 'exome'
+    mt_path = ''
+    for analysis in result['project']['analyses']:
+        if analysis['output'].endswith('.mt') and (
+            (seq_type_exome and '/exome/' in analysis['output'])
+            or (not seq_type_exome and '/exome/' not in analysis['output'])
+        ):
+            mt_path = analysis['output']
+
+    if not mt_path:
+        raise ValueError(f'No MT found for dataset {dataset}')
+    return mt_path
+
+
+@stage(required_stages=[QueryPanelapp])
+class RunHailFiltering(DatasetStage):
+    """
+    hail job to filter & label the MT
+    """
+
+    def expected_outputs(self, dataset: Dataset) -> dict[str, Path]:
+        return {
+            'labelled_vcf': dataset.prefix() / DATED_FOLDER / 'hail_labelled.vcf.bgz',
+            'pedigree': dataset.prefix() / DATED_FOLDER / 'pedigree.ped',
+        }
+
+    def queue_jobs(self, dataset: Dataset, inputs: StageInput) -> StageOutput:
+
+        # either do as you're told, or find the latest
+        # this could potentially follow on from the AnnotateDataset stage
+        # if this becomes integrated in the main pipeline
+        input_mt = get_config()['workflow'].get(
```

Comment:
This could technically load a different config than what was run. Not sure if that's intentional, but you could move the get_config() to the start and define as a constant